docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,525 @@
+
+== Paging
+
+You should only use `list` and `stream` methods if your collection contains small enough data sets. For larger data
+sets you can use the `find` method equivalents, which return a `PanacheQuery` on which you can do paging:
+
+[source,java]
+--
+// create a query for all living persons
+PanacheQuery<Person> livingPersons = Person.find("status", Status.Alive);
+
+// specify a projection to limit the fields returned from the database
+livingPersons.project("name", "birthdate");
+
+// make it use pages of 25 entries at a time
+livingPersons.page(Page.ofSize(25));
+
+// get the first page
+List<Person> firstPage = livingPersons.list();
+
+// get the second page
+List<Person> secondPage = livingPersons.nextPage().list();
+
+// get page 7
+List<Person> page7 = livingPersons.page(Page.of(7, 25)).list();
+
+// get the number of pages
+int numberOfPages = livingPersons.pageCount();
+
+// get the total number of entities returned by this query without paging
+int count = livingPersons.count();
+
+// and you can chain methods of course
+return Person.find("status", Status.Alive)
+    .page(Page.ofSize(25))
+    .nextPage()
+    .stream()
+--
+
+The `PanacheQuery` type has many other methods to deal with paging and returning streams.
+
+== Sorting
+
+All methods accepting a query string also accept an optional `Sort` parameter, which allows you to abstract your sorting:
+
+[source,java]
+--
+List<Person> persons = Person.list(Sort.by("name").and("birth"));
+
+// and with more restrictions
+List<Person> persons = Person.list("status", Sort.by("name").and("birth"), Status.Alive);
+--
+
+The `Sort` class has plenty of methods for adding columns and specifying sort direction.
+
+== Adding entity methods
+
+In general, we recommend not adding custom queries for your entities outside of the entities themselves,
+to keep all model queries close to the models they operate on. So we recommend adding them as static methods
+in your entity class:
+
+[source,java]
+--
+public class Person extends PanacheMongoEntity {
+    public String name;
+    public LocalDate birth;
+    public Status status;
+
+    public static Person findByName(String name){
+      return find("name", name).firstResult();
+    }
+
+    public static List<Person> findAlive(){
+      return list("status", Status.Alive);
+    }
+
+    public static void deleteLoics(){
+      delete("name", "Loïc");
+    }
+}
+--
+
+== Simplified queries
+
+Normally, MongoDB queries are of this form: `{'firstname': 'John', 'lastname':'Doe'}`, this is what we call MongoDB native queries.
+
+You can use them if you want, but we also support what we call **PanacheQL** that can be seen as a subset of link:https://docs.oracle.com/javaee/7/tutorial/persistence-querylanguage.htm#BNBTG[JPQL] (or link:https://docs.jboss.org/hibernate/core/3.3/reference/en/html/queryhql.html[HQL]) and allows you to easily express a query.
```

Comment:
Let's point to something more recent that Hibernate ORM 3.3 :)

https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#hql

Reply:
done
first result from google :)